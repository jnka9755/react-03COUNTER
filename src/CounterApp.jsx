// export function App() {

//     return <h1>Hola Mundo</h1>;
// }

import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value );

  const sum = () => {
    // setCounter( counter + 1 );
    setCounter ( (c) => c + 1);
  }

  const subtract = () => {

    setCounter ( (c) => c - 1);
  }

  const reset = () => {
    
    setCounter ((c) => c = value);
  }

  return (
    <>
      <h1> CounterApp </h1>
      <h2> { counter } </h2>

      <button onClick={ sum }> +1 </button>
      <button onClick={ subtract }> -1 </button>
      <button onClick={ reset }> Reset </button>
    </>
  )
}

CounterApp.protoTypes ={
  value: PropTypes.number.isRequired
}