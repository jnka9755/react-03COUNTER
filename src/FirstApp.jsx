import PropTypes from 'prop-types'

const name = 'Juan Camilo';

const getName = ( name ) => {
    return `Hola, soy ${ name }`;
}

export const FirstApp = ({ title, subtitle }) => {

  return (
    <>
        {/* <span> { getName('Juan') } </span> */}
        <h1 data-testid="test-title" >{ title }</h1>
        {/* <p>{ subtitle + 25 }</p> */}
        <p>{ subtitle  }</p>
        <p>{ subtitle  }</p>
        <p>{ subtitle  }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number
}

FirstApp.defaultProps = {
  title: 'Titulo por defecto',
  subtitle: 0
}