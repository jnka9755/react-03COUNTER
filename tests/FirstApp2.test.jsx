import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en FirstApp', () => { 

  const title = 'Soy una titulo prueba';
  const subtitle = 1234;

    test('Evaluar cambios en el componente', () => { 
        
        const { container } = render (
          <FirstApp 
            title = { title }
          />
        );

        expect( container ).toMatchSnapshot();
    });

    test('Evalua el titulo del componente', () => { 

      render (
        <FirstApp 
          title = { title }
        />
      );

      expect( screen.getByText( title ) ).toBeTruthy();
    });


    test('Evalua el titulo en una h1', () => { 

      render (
        <FirstApp 
          title = { title }
        />
      );

      expect( screen.getByRole( 'heading' , { level: 1 }).innerHTML ).toContain( title );
    });

    test('Evalua el subtitulo enviado por props', () => {

      render (
        <FirstApp 
          title = { title }
          subtitle={ subtitle }
        />
      );

      expect( screen.getAllByText( subtitle ).length ).toBe(3);
    });
 });