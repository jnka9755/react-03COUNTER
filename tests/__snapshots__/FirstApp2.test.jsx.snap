// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Evaluar cambios en el componente 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy una titulo prueba
  </h1>
  <p>
    0
  </p>
  <p>
    0
  </p>
  <p>
    0
  </p>
</div>
`;
